cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -794,7 +838,7 @@ executeQuery era cModeP localNodeConnInfo q = do
    execQuery :: IO (Either AcquireFailure (Either EraMismatch result))
    execQuery = queryNodeLocalState localNodeConnInfo Nothing q
 
-getSbe :: CardanoEraStyle era -> ExceptT ShelleyQueryCmdError IO (ShelleyBasedEra era)
+getSbe :: Monad m => CardanoEraStyle era -> ExceptT ShelleyQueryCmdError m (ShelleyBasedEra era)
```

Comment:
Can we leave this as concrete `IO`?

Reply:
Fixed

Reply:
I had to back out that change.

There is one place where `m /= IO`, which is here: https://github.com/input-output-hk/cardano-node/blob/7438366f3c2dbb845bc7a73aee4288be23448b29/cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs#L691

Reply:
This was the compile error:

```
src/Cardano/CLI/Shelley/Run/Query.hs:691:18: error:
    • Couldn't match type 'IO'
                     with 'LocalStateQueryExpr
                             (BlockInMode mode) ChainPoint (QueryInMode mode) () IO'
      Expected type: ExceptT
                       ShelleyQueryCmdError
                       (LocalStateQueryExpr
                          (BlockInMode mode) ChainPoint (QueryInMode mode) () IO)
                       (ShelleyBasedEra era)
        Actual type: ExceptT
                       ShelleyQueryCmdError IO (ShelleyBasedEra era)
    • In a stmt of a 'do' block: sbe <- getSbe $ cardanoEraStyle era
      In the expression:
        do sbe <- getSbe $ cardanoEraStyle era
           firstExceptT ShelleyQueryCmdEraMismatch . ExceptT
             $ queryExpr . QueryInEra eInMode . QueryInShelleyBasedEra sbe
                 $ QueryStakePools
      In a case alternative:
          Just eInMode
            -> do sbe <- getSbe $ cardanoEraStyle era
                  firstExceptT ShelleyQueryCmdEraMismatch . ExceptT
                    $ queryExpr . QueryInEra eInMode . QueryInShelleyBasedEra sbe
                        $ QueryStakePools
    • Relevant bindings include
        eInMode :: EraInMode era mode
          (bound at src/Cardano/CLI/Shelley/Run/Query.hs:690:14)
        cMode :: ConsensusMode mode
          (bound at src/Cardano/CLI/Shelley/Run/Query.hs:687:11)
        localNodeConnInfo :: LocalNodeConnectInfo mode
          (bound at src/Cardano/CLI/Shelley/Run/Query.hs:678:7)
        cModeParams :: ConsensusModeParams mode
          (bound at src/Cardano/CLI/Shelley/Run/Query.hs:674:44)
    |
691 |           sbe <- getSbe $ cardanoEraStyle era
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```